win32ss/user/ntuser/kbdlayout.c

```diff
@@ -995,12 +996,34 @@ NtUserGetKeyboardLayoutName(
     return bRet;
 }
 
+/* Win: xxxImmLoadLayout */
+PIMEINFOEX APIENTRY co_UserImmLoadLayout(HKL hKL)
+{
+    PIMEINFOEX piiex;
+
+    if (!IS_IME_HKL(hKL) && !IS_CICERO_MODE())
+        return NULL;
+
+    piiex = ExAllocatePoolWithTag(PagedPool, sizeof(IMEINFOEX), USERTAG_IME);
+    if (!piiex)
+        return NULL;
+
+    if (!co_ClientImmLoadLayout(hKL, piiex))
+    {
+        ExFreePoolWithTag(piiex, USERTAG_IME);
+        return NULL;
+    }
+
+    return piiex;
+}
+
 /*
  * NtUserLoadKeyboardLayoutEx
  *
  * Loads keyboard layout with given locale id
  *
  * NOTE: We adopt a different design from Microsoft's one for security reason.
+ *       We don't use the 1st and 3rd parameters of NtUserLoadKeyboardLayoutEx.
```

Comment:
Do you know what they are for, on Windows?

Reply:
The 1st parameter is a kernel file handle.
I don't know what the 3rd parameter is.